### PR TITLE
parany: port to dune

### PR DESCRIPTION
**Pull request by mseri**
I was playing around with it and it was convenient to be able to vendor it. This should fix https://github.com/UnixJunkie/parany/issues/25

Note that I needed to add `threads` in https://github.com/UnixJunkie/parany/compare/master...mseri:dunify?expand=1#diff-d218652a79a651b9be8eee7641ea0893R4 otherwise there is an error linking `netsys`. I think the issue is with `netsys` but I guess it is better to keep the threads there in any case.

Signed-off-by: Marcello Seri <marcello.seri@gmail.com>

### CODE REVIEW

File: tests/dune [hunk 1]
```diff
@@ -0,0 +1,11 @@
+(executable
+ (name test)
+ (libraries parany unix)
+)
+
+(alias
+ (name runtest)
+ (action
+  (run ./test.exe 6 25)
```
Comment (mseri) [inline tests/dune]: Numbers here were completely arbitrary. I don't know if you have a better suggestion

**Comment by UnixJunkie** [comment]
I have a few comments concerning this PR:
- is this absolutely necessary that we have this dune-project file?
  I don't like when the build information is scattered.
  Can't this be a flag in the jbuild file?
- please add the description to the opam file, since you updated that file.
- I don't want two dune files; again, don't scatter the build information.
So, dune is the new name of the jbuild file?

**Comment by mseri** [comment]
Thanks for the comment. 

- `dune` is the stable version of `jbuilder` after it had to change name.

- The `(dune-project)` file defines the root of a dune project and allows you to set global configuration options if needed, it has to be there. If you omit it, dune will regenerate it at each invocation.

- The structure I used for the project is the idiomatic one, with sources and tests separated. In dune project each component has a separated dune file with the build instructions. Moving everything back to a single folder is non idiomatic but is possible, the downside in my opinion is that the dune files will become needlessly uglier and more involved as the modules will have to be listed and separated properly to avoid double rules generation.

I am going to add the description in the next commit.

**Comment by UnixJunkie** [comment]
I'll merge your PR, so thanks for it, but then I will change back some things to how I like them, even if this is non idiomatic.

**Comment by UnixJunkie** [comment]
If you are interested by another issue, there is this one which has the potential to increase parallelization performance: https://github.com/UnixJunkie/parany/issues/23
We'll have to get some C code from Parmap I think.

**Comment by mseri** [comment]
Thanks! I’ll have a look as I have a moment

**Comment by mseri** [comment]
Y outting everything together and removing the tests stanza, test has become a  module of the parany lobrsry. Are you sure you don’t want to keep the test executsble (and runtest alias) anyways?

**Comment by UnixJunkie** [comment]
If test is a module of the library, the it should be a private module.
I have kept the test exe and runtest alias, what do you mean?

**Comment by mseri** [comment]
Deleting `test/dune` when moving everything in src, you removed the possibility of running `dune runtest`. You should update the `src/dune` file as follows:
```
(library
 (name parany)
 (public_name parany)
 (modules parany)
 (libraries unix threads netmulticore netsys)
)
executable
 (name test)
 (modules test)
 (libraries parany unix)
)

(alias
 (name runtest)
 (action
  (run ./test.exe 6 25)
 )
)
```

Otherwise the test executable is a module called `Parany__Test` that is installed alongside parany.
With this you can also re-introduce the tests inside the opam file

**Comment by UnixJunkie** [comment]
you should checkout the master branch, pull, and look at the src/dune file again.
As far as I can tell, this file is fine:
```
(library
  (name parany)
  (public_name parany)
  (modules parany)
  (libraries unix threads netmulticore netsys))

(executable
  (name test)
  (modules test)
  (libraries parany unix))

(alias
  (name runtest)
  (action
  (run ./test.exe 16 800)))
```
Also, there is this in the opam file:
```
 ["dune" "runtest" "-p" name] {with-test}
```
I hust moved some of your contributions, they were not deleted blindly.

**Comment by UnixJunkie** [comment]
There is no Parany.Test installed anymore when I opam pin parany to the master.
If the problem you see persists, please open a PR.

**Comment by UnixJunkie** [comment]
'dune runtest' works fine also

**Comment by mseri** [comment]
Sure. That is correct. Maybe it is a cache issue. I see the change by git pulling but they don’t appear when I navigate the file using github

**Comment by mseri** [comment]
Sorry for the fuss

**Pull request merged**